### PR TITLE
Decode equal sign when parsing query parameters of figure insert tag

### LINKED ISSUE #4242: Inserttag "figure" is not creating responsive image code
### Affected version(s)

4.12.x, 4.13.x

### Description

Using the inserttag "figure" ignores image sizes. The output is plain html image with full resolution but it should be responsive picture in different versions.
```
{{picture::34bc0ef2-949e-11ec-952f-001a4a050652?size=40}}
{{figure::34bc0ef2-949e-11ec-952f-001a4a050652?size=40}}
```

When i´m using the same image with picture tag, the output is fine.

**HTML Picture Tag using image settings ID40**
```
<picture>
	<source srcset="assets/images/9/terminkalender-legende-7807f2b6.webp 510w, assets/images/2/terminkalender-legende-cf3b6d07.webp 1020w, assets/images/c/terminkalender-legende-44f616ab.webp 1530w, assets/images/1/terminkalender-legende-42191382.webp 1766w" sizes="100vw" media="(max-width: 768px)" type="image/webp" width="510" height="159">
	<source srcset="assets/images/3/terminkalender-legende-7897653c.png 510w, assets/images/3/terminkalender-legende-f1a63000.png 1020w, assets/images/8/terminkalender-legende-66e09df7.png 1530w, assets/images/5/terminkalender-legende-faac7944.png 1766w" sizes="100vw" media="(max-width: 768px)" width="510" height="159">
	<source srcset="assets/images/0/terminkalender-legende-e11c7fab.webp 480w, assets/images/e/terminkalender-legende-b0d899c0.webp 960w, assets/images/5/terminkalender-legende-8d0e2f4e.webp 1440w, assets/images/1/terminkalender-legende-42191382.webp 1766w" sizes="100vw" media="(max-width: 480px)" type="image/webp" width="480" height="150">
	<source srcset="assets/images/f/terminkalender-legende-c13c3727.png 480w, assets/images/0/terminkalender-legende-6939d3c3.png 960w, assets/images/c/terminkalender-legende-c7bd677c.png 1440w, assets/images/5/terminkalender-legende-faac7944.png 1766w" sizes="100vw" media="(max-width: 480px)" width="480" height="150">
	<source srcset="assets/images/d/terminkalender-legende-81719635.webp 690w, assets/images/a/terminkalender-legende-816a6a8c.webp 1380w, assets/images/1/terminkalender-legende-42191382.webp 1766w" sizes="100vw" media="(max-width: 1200px)" type="image/webp" width="690" height="216">
	<source srcset="assets/images/8/terminkalender-legende-d9c9e38f.png 690w, assets/images/e/terminkalender-legende-20509abc.png 1380w, assets/images/5/terminkalender-legende-faac7944.png 1766w" sizes="100vw" media="(max-width: 1200px)" width="690" height="216">
	<source srcset="assets/images/f/terminkalender-legende-7cb6c01e.webp 870w, assets/images/4/terminkalender-legende-82f00e51.webp 1740w, assets/images/1/terminkalender-legende-42191382.webp 1766w" sizes="100vw" type="image/webp" width="870" height="272">
    
<img src="assets/images/7/terminkalender-legende-d4f2e4bb.png" srcset="assets/images/7/terminkalender-legende-d4f2e4bb.png 870w, assets/images/b/terminkalender-legende-5da008e7.png 1740w, assets/images/5/terminkalender-legende-faac7944.png 1766w" sizes="100vw" width="870" height="272" alt="" loading="lazy" class="img-fluid">
</picture>
```

**HTML Figure Tag using image settings ID40**
`<figure class="image_container"></figure>`

**Settings from images ID40**
Bildschirmfoto 2022-03-07 um 11 14 30

Confirmed in Contao 4.12.7 and 4.13

Greetz
Frank

### PR DESCRIPTION
**Pull request by m-vo**
Fixes #4242 

@Kahmoon Can you check if this fixes your issue?

### CODE REVIEW

File: core-bundle/src/Resources/contao/library/Contao/InsertTags.php [hunk 1]
```diff
@@ -1286,7 +1286,7 @@ static function ($allowedTag)
 	private function parseUrlWithQueryString(string $url): array
 	{
 		// Restore [&] and &amp;
-		$url = str_replace(array('[&]', '&amp;'), '&', $url);
+		$url = str_replace(array('[&]', '&amp;', '&#61;'), array('&', '&', '='), $url);
```
Comment (ausi) [inline core-bundle/src/Resources/contao/library/Contao/InsertTags.php]:
```suggestion
		$url = str_replace(array('&#61;', '[&]', '&amp;'), array('=', '&', '&'), $url);
```
As `&amp;#61;` should not get decoded to `=` I think?

**Review by ausi — APPROVED** [review]

**Comment by Kahmoon** [comment]
> Fixes #4242
> 
> @Kahmoon Can you check if this fixes your issue?

I Contao 4.12.7 the line 1289 is different from 4.13. I don´t have a 4.13 installation yet.

**Comment by fritzmg** [comment]
@Kahmoon you need to test with Contao 4.13. Contao 4.12 has reached the end of its life.

**Comment by Kahmoon** [comment]
> @Kahmoon you need to test with Contao 4.13. Contao 4.12 has reached the end of its life.

I don´t have a 4.13 yet (due incomptible extensions) until they are updated, which will probably take a while ;)

**Comment by ausi** [comment]
> I Contao 4.12.7 the line 1289 is different from 4.13.

Replace this line: https://github.com/contao/contao/blob/27c7730e8f3b7cb26fb2aa646ac17d5b99a1031c/core-bundle/src/Resources/contao/library/Contao/InsertTags.php#L1206-L1207

With this one:

```php
$url = str_replace(array('&#61;', '[&]', '&amp;'), array('=', '&', '&'), $url);
```

**Comment by Kahmoon** [comment]
yep, this fix is working as far as i can see 

Lightbox is not working but i m not sure if this is another problem. Code looks fine imho

```
<figure class="image_container">
<a href="assets/images/5/terminkalender-legende-faac7944.png" title="" data-lightbox="" class="cboxElement">
<picture>
<source srcset="assets/images/9/terminkalender-legende-7807f2b6.webp 510w, assets/images/2/terminkalender-legende-cf3b6d07.webp 1020w, assets/images/c/terminkalender-legende-44f616ab.webp 1530w, assets/images/1/terminkalender-legende-42191382.webp 1766w" sizes="100vw" media="(max-width: 768px)" type="image/webp" width="510" height="159">
<source srcset="assets/images/3/terminkalender-legende-7897653c.png 510w, assets/images/3/terminkalender-legende-f1a63000.png 1020w, assets/images/8/terminkalender-legende-66e09df7.png 1530w, assets/images/5/terminkalender-legende-faac7944.png 1766w" sizes="100vw" media="(max-width: 768px)" width="510" height="159">
<source srcset="assets/images/0/terminkalender-legende-e11c7fab.webp 480w, assets/images/e/terminkalender-legende-b0d899c0.webp 960w, assets/images/5/terminkalender-legende-8d0e2f4e.webp 1440w, assets/images/1/terminkalender-legende-42191382.webp 1766w" sizes="100vw" media="(max-width: 480px)" type="image/webp" width="480" height="150">
<source srcset="assets/images/f/terminkalender-legende-c13c3727.png 480w, assets/images/0/terminkalender-legende-6939d3c3.png 960w, assets/images/c/terminkalender-legende-c7bd677c.png 1440w, assets/images/5/terminkalender-legende-faac7944.png 1766w" sizes="100vw" media="(max-width: 480px)" width="480" height="150">
<source srcset="assets/images/d/terminkalender-legende-81719635.webp 690w, assets/images/a/terminkalender-legende-816a6a8c.webp 1380w, assets/images/1/terminkalender-legende-42191382.webp 1766w" sizes="100vw" media="(max-width: 1200px)" type="image/webp" width="690" height="216">
<source srcset="assets/images/8/terminkalender-legende-d9c9e38f.png 690w, assets/images/e/terminkalender-legende-20509abc.png 1380w, assets/images/5/terminkalender-legende-faac7944.png 1766w" sizes="100vw" media="(max-width: 1200px)" width="690" height="216">
<source srcset="assets/images/f/terminkalender-legende-7cb6c01e.webp 870w, assets/images/4/terminkalender-legende-82f00e51.webp 1740w, assets/images/1/terminkalender-legende-42191382.webp 1766w" sizes="100vw" type="image/webp" width="870" height="272">

<img src="assets/images/7/terminkalender-legende-d4f2e4bb.png" alt="" srcset="assets/images/7/terminkalender-legende-d4f2e4bb.png 870w, assets/images/b/terminkalender-legende-5da008e7.png 1740w, assets/images/5/terminkalender-legende-faac7944.png 1766w" sizes="100vw" width="870" height="272" loading="lazy" class="img-fluid">
</picture>
</a>
</figure>
```

**Comment by m-vo** [comment]
> Lightbox is not working but i m not sure if this is another problem

How's your insert tag looking in that case?

**Comment by Kahmoon** [comment]
> > Lightbox is not working but i m not sure if this is another problem
> 
> How's your insert tag looking in that case?

Nvm, its ok. Probably caching thing

`{{figure::34bc0ef2-949e-11ec-952f-001a4a050652?size=40[&]enableLightbox=1}}`

**Comment by leofeyer** [comment]
Thank you @m-vo.

**Pull request merged**